### ISSUE TITLE
Feature Request: Smarter queries

### ISSUE THREAD

**Issue by QuotableWater7** [Automattic/mongoose#4873]
I would like to chat about possibly removing some boilerplate around queries, by accepting either an array of values or a single value for a given query key.

**What is the current behavior?**
```js
someModel.find({ someKey: oneValue })

// OR

someModel.find({
  someKey: { $in: arrayOfValues },
})
```

**What is the desired behavior?**
```js
someModel.find({  someKey: oneValue })

// OR

someModel.find({ someKey: arrayOfValues })
```

**Why?**
I worked previously with ActiveRecord, and they had abstracted the difference between single value and an array of values into the ORM.  It makes for _slightly_ cleaner code, as the nested braces can be difficult to read on the same line.  We often take a simple query and spread it across 3 lines just to avoid that.

I wanted to see if the `mongoose` maintainers would be open to this sort of feature -- if so, I am more than happy to write the code that can make this happen.  I might have a question or two about the likely spots I'd need to hit, though.

Also, to clarify, this would be 100% backwards compatible.

Thanks for your time!

**Comment by TrejGun** [comment]
good point for 5.0

**Comment by QuotableWater7** [comment]
@TrejGun Just wanted to check in, are there any next-steps from my side (getting more decision makers to take a look, submitting a PR, etc)?  Thanks!

**Comment by TrejGun** [comment]
actually, looking at the code more close it seems you can use array operations on single element
https://github.com/Automattic/mongoose/blob/master/lib/schematype.js#L881

howeven it will create a query like
```
{ _id: { '$in': [ ObjectId("5875a365589f0c2d29f0c2e6") ] } }
```
but not
```
{ _id: { '$eq': [ ObjectId("5875a365589f0c2d29f0c2e6") ] } }
```

I'm pretty sure @vkarpov15 won't merge this in mongoose 4, so if you want to switch operator at runtime you probably should try to write a plugin

**Comment by QuotableWater7** [comment]
@TrejGun Correct me if I'm wrong, but I think your example is slightly different than I'm going for.  I'm basically wanting to do:
```js
{ _id: someVariableContainingArrayOfIds }
```

Whereas right now, I'm forced to expand out some boilerplate:
```js
{ _id: { $in: someVariableContainingArrayOfIds } } 
```

Does that make sense?

**Comment by vkarpov15** [comment]
I really like this idea, definitely something to put in for a future release. Would you be interested in putting in a PR @QuotableWater7 ? I'd be happy to help, otherwise gonna have to wait until I start working on 4.9.

**Comment by QuotableWater7** [comment]
Cool, thanks for taking a look @vkarpov15!  I would absolutely be happy to put together a PR -- I'll try and acquaint myself with the code tonight.

**Comment by QuotableWater7** [comment]
@vkarpov15 I opened a PR to get some feedback on the approach.  Thanks!
https://github.com/Automattic/mongoose/pull/4912

**Comment by vkarpov15** [comment]
Fixed by #4912
